src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -24157,18 +24321,104 @@ interface I1
         [Fact]
         public void IsPattern_07()
         {
-            var src = @"
+            var src1 = @"
 class Helper1<T, U>
```

Comment:
Are we testing the following where `U` has `allows ref struct`?
```csharp
class Helper1<T, U>
    where U : allows ref struct
{
    public static void Test1(U h1)
    {
        if (h1 is T t)
        ...
    }
}

Helper1<I1, S>.Test1(new S());
...
```

Reply:
Same response